pkg/ingester/ingester.go

```diff
@@ -374,9 +389,25 @@ func (i *Ingester) Push(ctx context.Context, req *client.WriteRequest) (*client.
 		}
 	}
 
-	if len(req.Metadata) > 0 {
+	var firstMetadataErr error
+	for _, metadata := range req.Metadata {
+		err := i.appendMetadata(ctx, userID, metadata)
```

Comment:
Since this is user-supplied data, can `metadata` be nil at this point? If so, `appendMetadata` will panic.

Reply:
I have started using the `Get` prefixed functions all around, thanks for the hint in that direction 😄